### DISCUSSION TITLE
type="file" localisation

### DISCUSSION THREAD

**Discussion by Aporrhia** [moonshine-software/moonshine#1278] [Ideas]
Current disadvantage is that it is NOT possible to change localisation for `type="file"`. 

It would be great to change it by the _locale_ of the website.
In instance:
By using `app()->getLocale();` it will be possible to set it to the _locale_ of the app.

**Comment by lee-to** [comment]
and where exactly is there no localization?

**Reply by Mr-Coquin** [reply]
> and where exactly is there no localization?

Я думаю он имеет виду, что если стоит EN а на компе RU язык то будет вот такая картина

![image](https://github.com/user-attachments/assets/08ab67bc-51a0-4a8b-95fa-7a4c259ab381)

**Reply by Aporrhia** [reply]
Yes, due to the attribute's dependence on the OS and browser settings, as DissNik said, the language may differ from the language the site itself is based on, which is in some cases a so-called problem.

**Comment by DissNik** [comment]
This is a system element that depends on the OS and browser settings.

**Reply by Mr-Coquin** [reply]
Это понятно. Но было бы не плохо задать свой текст кстати добавив что то такого в шаблон поля 

```
<style>
  .file-input-container {
    overflow: hidden;
    position: relative;
  }
  
  .file-input-label {
    background: #f0f0f0;
    padding: 8px 15px;
    cursor: pointer;
    border-radius: 5px;
    border: 1px solid #d3d3d3;
  }
  
  .file-input {
    width: 0.1px;
    height: 0.1px;
    opacity: 0;
    position: absolute;
    z-index: -1;
  }
</style>

<div class="file-input-container">
  <label class="file-input-label" for="file-upload">Загрузить файл</label>
  <input id="file-upload" class="file-input" type="file">
</div>
```

**Reply by DissNik** [reply]
но в таком случае мы потеряем перечень загружаемых файлов который есть в стандартном инпуте

**Reply by Mr-Coquin** [reply]
> но в таком случае мы потеряем перечень загружаемых файлов который есть в стандартном инпуте

Ну это так на быстро что смог найти как решение.  Но понимаю ситуацию, у самого недавно стал вопрос , заказчик ругается. У его сайт на EN а комп на RU и все баг и исправляй.  Хочу форму на языке интерфейса сайта а не моего компа :)

**Reply by DissNik** [reply]
никто не запрещает переопределять вьюху у поля и ваше решение как раз в таких случаях будет в самый раз

**Reply by Mr-Coquin** [reply]
> но в таком случае мы потеряем перечень загружаемых файлов который есть в стандартном инпуте

но в таком случае мы потеряем перечень загружаемых файлов который есть в стандартном инпуте  :)

**Reply by DissNik** [reply]
так это затронет только вас, а не всех кто использует муншайн в своих проектах )

**Reply by DissNik** [reply]
Я про то, что такое по дефолту не стоит делать. А то, что вы кастомизируете, то это нормально для каждого конкретного случая.
